ci/test.lua

```diff
@@ -415,12 +482,22 @@ local function filter_tests(tests, config)
     return status
 end
 
-local function run_tests(tests, status, counts)
+local function run_tests(tests, status, counts, config)
     print(('Running %d tests'):format(#tests))
     for _, test in pairs(tests) do
-        MODE_NAVIGATE_FNS[test.config.mode]()
-        local passed = run_test(test, status, counts)
-        status[test.full_name] = passed and TestStatus.PASSED or TestStatus.FAILED
+        status[test.full_name] = TestStatus.FAILED
+        if MODES[test.config.mode].failed then goto skip end
```

Comment:
Is this desired behavior? Unrunnable tests are recorded as failed? Or should they just not be recorded at all?

Reply:
The current approach wouldn't print an error for each test, right? My understanding is that saving a `FAILED` status to the status file would just prevent the test from re-running again on the next DF run.

Actually, if status for these tests remain `PENDING`, leaving them untouched would allow for them to run again on future DF runs, which might be a reasonable behavior (e.g. if DF gets stuck on a screen for some reason and the issue goes away when DF restarts).

Reply:
done. I also printed a line for how many tests were in unreachable modes. otherwise they would be completely silently skipped